.build/build-owasp.xml

```diff
@@ -52,35 +72,125 @@
+                </outputfilterchain>
+            </redirector>
+        </exec>
+        <echo message="last.build.number=${last.build.number}"/>
+
+        <property name="last.build.url"
+                  value="${nightlies.url}/${last.build.number}/Cassandra-${nightly.version}-artifacts/jdk=jdk_1.8_latest,label=cassandra/build"/>
+        <exec executable="curl" failonerror="true" failifexecutionfails="true" logerror="true">
+            <arg value="-s"/>
+            <arg value="${last.build.url}/"/>
+            <redirector outputproperty="last.build.name">
+                <outputfilterchain>
+                    <linecontainsregexp regexp="-bin\.tar\.gz"/>
+                    <replaceregex byline="true" pattern=".*href=&quot;([^&quot;]+)&quot;.*" replace="\1"/>
+                    <headfilter lines="1"/>
+                </outputfilterchain>
+            </redirector>
+        </exec>
+        <property name="last.build.artifact.url" value="${last.build.url}/${last.build.name}"/>
+        <echo message="last.build.artifact.url=${last.build.artifact.url}"/>
+    </target>
+
+    <target name="-maybe-fetch-latest-nightly-build" depends="-resolve-latest-nightly-build" unless="nightly.build.dir">
+        <property name="cassandra.nightlies.dir" value="${tmp.dir}/cassandra-nightlies"/>
+        <property name="nightly.build.dir" value="${cassandra.nightlies.dir}/${nightly.version}/${last.build.number}"/>
+        <condition property="nightly.build.dir.exists">
+            <available file="${nightly.build.dir}" type="dir"/>
+        </condition>
+        <antcall target="-fetch-latest-nightly-build" unless:set="nightly.build.dir.exists" inheritall="true"
+                 inheritrefs="true"/>
+    </target>
+
+    <target name="-fetch-latest-nightly-build">
+        <dirname property="nightly.build.parent.dir" file="${nightly.build.dir}"/>
+        <delete dir="${nightly.build.parent.dir}" includeemptydirs="true" failonerror="false"/>
+        <mkdir dir="${nightly.build.dir}"/>
+        <get src="${last.build.artifact.url}" dest="${nightly.build.dir}/cassandra.tar.gz" verbose="true"/>
+        <gunzip src="${nightly.build.dir}/cassandra.tar.gz" dest="${nightly.build.dir}/cassandra.tar"/>
+        <delete file="${nightly.build.dir}/cassandra.tar.gz"/>
+        <untar src="${nightly.build.dir}/cassandra.tar" dest="${nightly.build.dir}/" allowfilestoescapedest="false"
+               stripabsolutepathspec="2"/>
+        <delete file="${nightly.build.dir}/cassandra.tar"/>
+        <exec executable="bash" failifexecutionfails="true" failonerror="true" logerror="true"
+              dir="${nightly.build.dir}">
+            <arg value="-c"/>
+            <arg value="mv apache-cassandra-*/* .; rm -rf apache-cassandra-*"/>
+        </exec>
+    </target>
+
+    <target name="-determine-dependency-changes">
+        <get src="https://raw.githubusercontent.com/apache/cassandra/cassandra-3.0/build.xml" retries="3"
```

Comment:
The branch is hardcoded, is it correct? Should we determine the branch based on the git command output results?

Reply:
no in this case, we want to compare with the upstream branch thus this will be different in each PR